### DISCUSSION TITLE
Stream url

### DISCUSSION THREAD

**Discussion by btc-asset-mgr** [sbooth/SFBAudioEngine#348] [General]
Dear Stephen,The freamwork  can not play the stream url like this 
`Download URL: https://cn-beijing-data.aliyundrive.net/9AdvKEM1%2F2407895630%2F66a5a8557b779a02d5fe4cad8283a287560b59e6%2F66a5a8557ffff7f7639e438cbad38cf9f6e001fa?callback=eyJjYWxsYmFja1VybCI6Imh0dHA6Ly9iajI5LmFwaS1ocC5hbGl5dW5wZHMuY29tL3YyL2ZpbGUvZG93bmxvYWRfY2FsbGJhY2siLCJjYWxsYmFja0JvZHkiOiJodHRwSGVhZGVyLnJhbmdlPSR7aHR0cEhlYWRlci5yYW5nZX1cdTAwMjZidWNrZXQ9JHtidWNrZXR9XHUwMDI2b2JqZWN0PSR7b2JqZWN0fVx1MDAyNmRvbWFpbl9pZD0ke3g6ZG9tYWluX2lkfVx1MDAyNnVzZXJfaWQ9JHt4OnVzZXJfaWR9XHUwMDI2ZHJpdmVfaWQ9JHt4OmRyaXZlX2lkfVx1MDAyNmZpbGVfaWQ9JHt4OmZpbGVfaWR9XHUwMDI2cGRzX3BhcmFtcz0ke3g6cGRzX3BhcmFtc31cdTAwMjZ2ZXJzaW9uPSR7eDp2ZXJzaW9ufSIsImNhbGxiYWNrQm9keVR5cGUiOiJhcHBsaWNhdGlvbi94LXd3dy1mb3JtLXVybGVuY29kZWQiLCJjYWxsYmFja1N0YWdlIjoiYmVmb3JlLWV4ZWN1dGUiLCJjYWxsYmFja0ZhaWx1cmVBY3Rpb24iOiJpZ25vcmUifQ%3D%3D&callback-var=eyJ4OmRvbWFpbl9pZCI6ImJqMjkiLCJ4OnVzZXJfaWQiOiI3ODEwMjAyZjBlODU0YzYwYTJjMzIyNDNmY2ZlMzQ0MyIsIng6ZHJpdmVfaWQiOiIyNDA3ODk1NjMwIiwieDpmaWxlX2lkIjoiNjZhNWE4NTU3Yjc3OWEwMmQ1ZmU0Y2FkODI4M2EyODc1NjBiNTllNiIsIng6cGRzX3BhcmFtcyI6IntcImFwXCI6XCI5ZGQxYTczOTk5YzA0NWJkYTcyMGEzMDViNmI1NDYxMVwifSIsIng6dmVyc2lvbiI6InYzIn0%3D&di=bj29&dr=2407895630&f=66a5a8557b779a02d5fe4cad8283a287560b59e6&pds-params=%7B%22ap%22%3A%229dd1a73999c045bda720a305b6b54611%22%7D&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%2703%2520-%2520Hurd%2520-%2520%25D0%259D%25D0%25B0%25D1%2580%2520%25D0%25A1%25D0%25B0%25D1%2580.flac&security-token=CAISvgJ1q6Ft5B2yfSjIr5fNMv7l24sV%2BYeOMVzm1VYnNfdegJznljz2IHhMf3NpBOkZvvQ1lGlU6%2Fcalq5rR4QAXlDfNRCIIF2nq1HPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLs0ARrv4r1fMUqW2X543AUgFLhy2KKMPY99xpFgh9a7j0iCbSGyUu%2FhcRm5sw9%2Byfo34lVYne47DQ0wsh7uHwufJ7FxfIREfquk63pvSlHLcLPe0Kjzzleo2k1XRPVFF%2B535IaHXuToXDnvSiW0m1AvXtuMkagAEP%2BVuCnjPlnMzH13rqdxrfBN0nVjhO0%2FIF2d1Tepd%2Bd5LhajvCnyODZxHp586Skcla8%2FbcMxWRY83oIpSrc05ViFE25msIXde70iBXgepXP%2B2Mu620qD6sWiwUFuGKz%2B8hzc5JZChPijNlqL88aAZkAVXMf5mv83EsjR2XhDU1WCAA&u=7810202f0e854c60a2c32243fcfe3443&x-oss-access-key-id=STS.NTxyDQ6T4NEe3zW1Vr9xroYLt&x-oss-expires=1724381668&x-oss-signature=Z7G0ymNTTaM3xP02k5Ei%2F3LhfCjnEJ0PaHM1pBSARHY%3D&x-oss-signature-version=OSS2
Stream URL: https://cn-beijing-data.aliyundrive.net/9AdvKEM1%2F2407895630%2F66a5a8557b779a02d5fe4cad8283a287560b59e6%2F66a5a8557ffff7f7639e438cbad38cf9f6e001fa?callback=eyJjYWxsYmFja1VybCI6Imh0dHA6Ly9iajI5LmFwaS1ocC5hbGl5dW5wZHMuY29tL3YyL2ZpbGUvZG93bmxvYWRfY2FsbGJhY2siLCJjYWxsYmFja0JvZHkiOiJodHRwSGVhZGVyLnJhbmdlPSR7aHR0cEhlYWRlci5yYW5nZX1cdTAwMjZidWNrZXQ9JHtidWNrZXR9XHUwMDI2b2JqZWN0PSR7b2JqZWN0fVx1MDAyNmRvbWFpbl9pZD0ke3g6ZG9tYWluX2lkfVx1MDAyNnVzZXJfaWQ9JHt4OnVzZXJfaWR9XHUwMDI2ZHJpdmVfaWQ9JHt4OmRyaXZlX2lkfVx1MDAyNmZpbGVfaWQ9JHt4OmZpbGVfaWR9XHUwMDI2cGRzX3BhcmFtcz0ke3g6cGRzX3BhcmFtc31cdTAwMjZ2ZXJzaW9uPSR7eDp2ZXJzaW9ufSIsImNhbGxiYWNrQm9keVR5cGUiOiJhcHBsaWNhdGlvbi94LXd3dy1mb3JtLXVybGVuY29kZWQiLCJjYWxsYmFja1N0YWdlIjoiYmVmb3JlLWV4ZWN1dGUiLCJjYWxsYmFja0ZhaWx1cmVBY3Rpb24iOiJpZ25vcmUifQ%3D%3D&callback-var=eyJ4OmRvbWFpbl9pZCI6ImJqMjkiLCJ4OnVzZXJfaWQiOiI3ODEwMjAyZjBlODU0YzYwYTJjMzIyNDNmY2ZlMzQ0MyIsIng6ZHJpdmVfaWQiOiIyNDA3ODk1NjMwIiwieDpmaWxlX2lkIjoiNjZhNWE4NTU3Yjc3OWEwMmQ1ZmU0Y2FkODI4M2EyODc1NjBiNTllNiIsIng6cGRzX3BhcmFtcyI6IntcImFwXCI6XCI5ZGQxYTczOTk5YzA0NWJkYTcyMGEzMDViNmI1NDYxMVwifSIsIng6dmVyc2lvbiI6InYzIn0%3D&di=bj29&dr=2407895630&f=66a5a8557b779a02d5fe4cad8283a287560b59e6&pds-params=%7B%22ap%22%3A%229dd1a73999c045bda720a305b6b54611%22%7D&response-content-disposition=attachment%3B%20filename%2A%3DUTF-8%27%2703%2520-%2520Hurd%2520-%2520%25D0%259D%25D0%25B0%25D1%2580%2520%25D0%25A1%25D0%25B0%25D1%2580.flac&security-token=CAISvgJ1q6Ft5B2yfSjIr5fNMv7l24sV%2BYeOMVzm1VYnNfdegJznljz2IHhMf3NpBOkZvvQ1lGlU6%2Fcalq5rR4QAXlDfNRCIIF2nq1HPWZHInuDox55m4cTXNAr%2BIhr%2F29CoEIedZdjBe%2FCrRknZnytou9XTfimjWFrXWv%2Fgy%2BQQDLItUxK%2FcCBNCfpPOwJms7V6D3bKMuu3OROY6Qi5TmgQ41Uh1jgjtPzkkpfFtkGF1GeXkLFF%2B97DRbG%2FdNRpMZtFVNO44fd7bKKp0lQLs0ARrv4r1fMUqW2X543AUgFLhy2KKMPY99xpFgh9a7j0iCbSGyUu%2FhcRm5sw9%2Byfo34lVYne47DQ0wsh7uHwufJ7FxfIREfquk63pvSlHLcLPe0Kjzzleo2k1XRPVFF%2B535IaHXuToXDnvSiW0m1AvXtuMkagAEP%2BVuCnjPlnMzH13rqdxrfBN0nVjhO0%2FIF2d1Tepd%2Bd5LhajvCnyODZxHp586Skcla8%2FbcMxWRY83oIpSrc05ViFE25msIXde70iBXgepXP%2B2Mu620qD6sWiwUFuGKz%2B8hzc5JZChPijNlqL88aAZkAVXMf5mv83EsjR2XhDU1WCAA&u=7810202f0e854c60a2c32243fcfe3443&x-oss-access-key-id=STS.NTxyDQ6T4NEe3zW1Vr9xroYLt&x-oss-expires=1724381668&x-oss-signature=Z7G0ymNTTaM3xP02k5Ei%2F3LhfCjnEJ0PaHM1pBSARHY%3D&x-oss-signature-version=OSS2
SFBAudioDecoder unsupported path extension:`,has an other way to play ? this is the chinese cloud aliyunpan,the final url,may xxx.flac

**Comment by sbooth** [comment]
The current version of `SFBAudioEngine` only supports file URLs. Previous (legacy) versions attempted to support http but the current library design is such that streaming would require a bit of a redesign on the input side.

You could perhaps create your own `SFBInputSource` subclass that performs downloading/buffering/etc but this isn't something I've tried.

**Reply by btc-asset-mgr** [reply]
has any example?

**Reply by sbooth** [reply]
No, there are no examples of this I know of.

**Reply by btc-asset-mgr** [reply]
Thank you sir,i wiil try it.

**Comment by btc-asset-mgr** [comment]
Dear Sir,And I have another question about the soud style,which part code can change sound style? expect the eq? I what it get more sound details, which code change can change it?

**Reply by sbooth** [reply]
You can add any number of `AVAudioUnit`s to the processing graph to achieve the sound you're after. Adding additional Audio Units is the same process as adding the equalizer.

**Comment by CharlesWiltgen** [comment]
Hey @sbooth, it turns out that I do actually need progressive streaming, so I'm about to create my own `SFBInputSource`. Has anyone ever done this, or am I the lucky one? 🙃

**Reply by sbooth** [reply]
> Has anyone ever done this, or am I the lucky one?

The number of people other than me (that I know of): 0 🤣

**Reply by CharlesWiltgen** [reply]
Quick update: I did this! There were a couple hitches, but it ultimately works well. Thank you!
